keps/sig-node/2411-cri-container-log-rotation/README.md

```diff
@@ -0,0 +1,301 @@
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The CRIContainerLogRotation feature gate was implemented in v1.10 and has been in Beta stage since v1.11. We would like to identify any gaps in the implementation of this feature so that we can promote it to stable as it has already been in production use for quite some time now. With this feature gate, the kubelet is in charge of managing the container log directory structure as well as rotating the logs when a certain (user configurable) limit is reached.
+
+## Motivation
+
+Container runtimes that communicate with the kubelet via the Container Runtime Interface (CRI) needed a container log management system. The kubelet was already in charge of determining the container log file path and passing that down to the container runtime so that it can write the container logs there. Thus making the kubelet in charge of rotating the container logs allows the kubelet to manage and access the logs directly without having to call the container runtime. An added advantage of this is that logging agents can ingest files directly without any further integrations with the container runtime.
+
+### Goals
+
+- The kubelet assigns the log path for a container and runtime writes the container output to that path
+- The kubelet periodically checks the disk space occupied by the container logs and rotates them if necessary
+- After the logs are rotated, the kubelet sends a signal to the container runtime to re-open the log file
+- The kubelet exposes a consistent log directory structure with metadata so that any logging agent can integrate with it
+
+### Non-Goals
+
+- Shipping the logs directly to a remote storage
+- Supporting container runtimes that run on a different virtual/physical machine from the kubelet
```

Comment:
let's also make it explicit non-goal to implement independent logs from containers management as described here:

https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/kubelet-cri-logging.md#user-content-requirements:~:text=Allow%20kubelet%20to%20manage%20the%20lifecycle,their%20logs%20need%20to%20be%20decoupled.

> Allow kubelet to manage the lifecycle of the logs to pave the way for better disk management in the future. This implies that the lifecycle of containers and their logs need to be decoupled.

Reply:
Done